includes/Core/Authentication/Setup.php

```diff
@@ -104,6 +105,22 @@ public function register() {
 		add_action( 'admin_action_' . Google_Proxy::ACTION_EXCHANGE_SITE_CODE, array( $this, 'handle_action_exchange_site_code' ) );
 	}
 
+	/**
+	 * Composes the oAuth proxy get help link.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The get help link.
+	 */
+	public function get_oauth_proxy_failed_help_link() {
```

Comment:
This helper method is only used within this class (when looking at the production code), and it's not apparent that it'll be needed outside of the class. Therefore I'd suggest it should not be `public` - I'd go for `protected` instead.

I know it's used in the unit tests, but it's generally not good practice to allow unit tests to dictate the public API of the code they are testing. Within the tests themselves I think testing for the string literal is fine in this case - and indeed, that will have the added benefit of actually verifting the _result_ of `get_oauth_proxy_failed_help_link()`.

```suggestion
	protected function get_oauth_proxy_failed_help_link() {
```